mesa/space.py

```diff
@@ -551,16 +588,24 @@ def position_agent(
             )
 
         if x == "random" or y == "random":
-            if len(self.empties) == 0:
+            if len(self._empties) == 0:
```

Comment:
`self._empties` isn't always defined.

Reply:
right, actually that piece of code was redundant because `move_to_empty` then check if the grid is full too, so I removed it